server/src/api/as/webhook.ts

```
@@ -0,0 +1,388 @@
+        // License downgraded to lower tier
+        // Note: Creates NEW license_key with prev_license_key pointing to old one
+        // AppSumo sends simultaneous deactivate event for old license (we skip it)
+        await handleDowngradeEvent(license_key, tier, prev_license_key);
+        break;
+
+      case "deactivate":
+        // License refunded or canceled
+        // Note: license_status is "active" in webhook (for refunds), becomes deactivated after our 200 response
+        // For upgrade/downgrade, license_status is already "deactivated" - we skip these
+        const isUpgradeOrDowngradeDeactivation =
+          extra?.reason === "Upgraded by customer" || extra?.reason === "Downgraded by customer";
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)